src/coreclr/jit/lowerxarch.cpp

```diff
@@ -173,6 +173,23 @@ GenTree* Lowering::LowerMul(GenTreeOp* mul)
 //
 GenTree* Lowering::LowerBinaryArithmetic(GenTreeOp* binOp)
 {
+    if (comp->opts.OptimizationEnabled() && binOp->OperIs(GT_XOR))
+    {
+        LIR::Use binOpUse;
+        GenTree* op1 = binOp->gtGetOp1();
+        GenTree* op2 = binOp->gtGetOp2();
+
+        if (op1->OperIs(GT_LCL_VAR) && op2->IsIntegralConst(0) && BlockRange().TryGetUse(binOp, &binOpUse))
+        {
+            binOpUse.ReplaceWith(op1);
+
+            BlockRange().Remove(op2);
+            BlockRange().Remove(binOp);
+
+            return op1;
+        }
+    }
```

Comment:
Consider moving to morph, e.g. `x*0 -> 0`: 
https://github.com/dotnet/runtime/blob/fe24ab3040b454fd04983aba9eb0f600a2fb43e9/src/coreclr/jit/morph.cpp#L13723-L13733

Reply:
in theory you can also improve this `x * 0` by not giving up on side effects and making a COMMA but it's up to you.
Just make sure you don't drop CSE candidates

Reply:
> in theory you can also improve this `x * 0` by not giving up on side effects and making a COMMA but it's up to you. Just make sure you don't drop CSE candidates

Could you give me more details about this? I'm a very newbie in the project and a lot of obvious things are not that obvious for me yet😊

Reply:
Don't worry, none of those things were obvious, :)